server/src/main/java/org/elasticsearch/common/logging/NodeNamePatternConverter.java

```diff
@@ -26,25 +26,39 @@
 import org.apache.logging.log4j.core.pattern.ConverterKeys;
 import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
 import org.apache.logging.log4j.core.pattern.PatternConverter;
-import org.apache.lucene.util.SetOnce;
 
 /**
  * Converts {@code %node_name} in log4j patterns into the current node name.
- * We *could* use a system property lookup instead but this is very explicit
- * and fails fast if we try to use the logger without initializing the node
- * name. As a bonus it ought to be ever so slightly faster because it doesn't
- * have to look up the system property every time.
+ * We can't use a system property for this because the node name system
+ * property is only set if the node name is explicitly defined in
+ * elasticsearch.yml.
  */
 @Plugin(category = PatternConverter.CATEGORY, name = "NodeNamePatternConverter")
 @ConverterKeys({"node_name"})
-public class NodeNamePatternConverter extends LogEventPatternConverter {
-    private static final SetOnce<String> NODE_NAME = new SetOnce<>();
+public final class NodeNamePatternConverter extends LogEventPatternConverter {
+    /**
```

Comment:
I'm not happy with this at all but it is another thing that I think we'll be able to remove it master in a follow up.

Reply:
@jasontedor could you take a look at this when you get a chance? I'm well aware it isn't good but I'm not clear on whether it is disastrous or simply "something we shouldn't keep around for long".

Reply:
I talked with @jasontedor and he'd prefer the volatile read over this so I'm switching back to it.